saas/backends/stripe_processor/views.py

```diff
@@ -79,17 +79,30 @@ class StripeWebhook(APIView):
     def post(self, request, *args, **kwargs):
         #pylint:disable=unused-argument,no-self-use
         from saas.models import Charge
+
         processor_backend = get_processor_backend(get_broker())
         stripe.api_key = processor_backend.priv_key
-        # Attempt to validate the event by posting it back to Stripe.
-        if django_settings.DEBUG:
-            event = stripe.Event.construct_from(request.DATA, stripe.api_key)
-        else:
-            event = stripe.Event.retrieve(request.DATA['id'])
+
+        endpoint_secret = django_settings.STRIPE_ENDPOINT_SECRET
+        payload = request.body
+        sig_header = request.META['HTTP_STRIPE_SIGNATURE']
+        event = None
+
+        try:
+            event = stripe.Webhook.construct_event(
+                payload, sig_header, endpoint_secret
+            )
+        except ValueError as e:
+            # Invalid payload
+            return Response(status=400)
+        except stripe.error.SignatureVerificationError as e:
+            # Invalid signature
+            return Response(status=400)
+
         event_type = event.type
         if not event:
```

Comment:
`if not event` together with use of `event.type` a few lines below seems like a issue.